pokecli.py

```diff
@@ -206,15 +211,20 @@ def main():
         bot.start()
 
         logger.log('[x] Starting PokemonGo Bot....', 'green')
+        idle = True
 
-        while True:
+        while idle:
             bot.take_step()
 
+        return
+
     except KeyboardInterrupt:
         logger.log('[x] Exiting PokemonGo Bot', 'red')
         # TODO Add number of pokemon catched, pokestops visited, highest CP
         # pokemon catched, etc.
 
+signal.signal(signal.SIGUSR2, reread_config)
 
 if __name__ == '__main__':
-    main()
+    while True:
```

Comment:
This is indeed weird. Please try to keep all necessary logic inside the `main` function (including loop) and remove this from the ` if __name__ == '__main__'` block. Make it as simple as possible, please.